.github/workflows/release.yml

```diff
@@ -9,7 +9,7 @@ on:
 
 jobs:
   create_release:
-    if: ${{ (github.event.workflow_run.conclusion == 'success') && startsWith(github.ref, 'refs/tags/v') }} # Skip if Rust workflow fails
+    if: ${{ (github.event.workflow_run.conclusion == 'success') && (startsWith(github.ref, 'refs/tags/v')) }} # Skip if Rust workflow fails
```

Comment:
This was the main error, but the difficulty in testing this without a "fake release" also led to adding the `check-actions` workflow, as otherwise we can run into small errors whenever needing to change CI and its a trivial addition to lint it.